demos/asp-net-core/Divergent.Sales.ViewModelComposition/OrderDetailsViewModelAppender.cs

```diff
@@ -0,0 +1,39 @@
+﻿using ITOps.ViewModelComposition;
+using ITOps.Json;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Divergent.Sales.ViewModelComposition
+{
+    public class OrderDetailsViewModelAppender : IViewModelAppender
+    {
+        public bool Matches(RouteData routeData, string verb)
+        {
+            /*
+             * matching is a bit weak in this sample, it's designed 
+             * this way to satisfy both the Gateway and the Mvc sample
+             */
+            var controller = (string)routeData.Values["controller"];
+
+            return HttpMethods.IsGet(verb)
+                && controller.ToLowerInvariant() == "orders"
+                && routeData.Values.ContainsKey("id");
+        }
+
+        public async Task Append(dynamic vm, RouteData routeData, IQueryCollection query)
+        {
+            var id = (string)routeData.Values["id"];
+
+            var url = $"http://localhost:20295/api/orders/{id}";
```

Comment:
it's super hard for a sample, I thought about it, the thing is the way .NET Core allows to inject (via DI) configuration options is not really nice and it might complicate a lot the sample. What if we accept this as is for now, raise an issue to track it and spike something later? so not to complicate even more this PR